taroscript/send.go

```diff
@@ -0,0 +1,163 @@
+package taroscript
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/lightninglabs/taro/asset"
+	"github.com/lightninglabs/taro/commitment"
+	"github.com/lightninglabs/taro/taropsbt"
+)
+
+var (
+	// ErrInvalidCollectibleSplit is returned when a collectible is split
+	// into more than two outputs.
+	ErrInvalidCollectibleSplit = errors.New(
+		"fund: invalid collectible split",
+	)
+
+	// ErrInvalidChangeOutputLocation is returned when the change output is
+	// not at the expected location (index 0).
+	ErrInvalidChangeOutputLocation = errors.New(
+		"fund: invalid change output location, should be index 0",
+	)
+
+	// ErrInvalidSplitAmounts is returned when the split amounts don't add
+	// up to the amount of the asset being spent.
+	ErrInvalidSplitAmounts = errors.New(
+		"fund: invalid split amounts, sum doesn't match input",
+	)
+)
+
+// PrepareOutputAssets prepares the assets of the given outputs depending on
+// the amounts set on the transaction. If a split is necessary (non-interactive
+// or partial amount send) it computes a split commitment with the given input
+// and spend information. The input MUST be checked as valid beforehand and the
+// change output is expected to be declared as such (and be at index 0).
+func PrepareOutputAssets(input *taropsbt.VInput,
```

Comment:
Cool yeh I think it makes sense to further unify them. We had to go back and make a few quick fixes as our understanding of the protocol evolved during the initial implementation.